### PR TITLE
Assign user to workspace on signin

### PR DESCRIPTION
**Pull request by charlesBochet**
When a user sign-in:
- triggers an event (based on users.created in Hasura)
- get this event in server
- fetch the corresponding workspace through prisma
- upsert a workspace_member

image

Then, on token generation:
- adds workspace-id in JWT payload

image

### CODE REVIEW

**Review by Samox — APPROVED** [review]

File: hasura/migrations/default/1682141433503_alter_table_public_workspace_members_add_column_created_at/down.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-- Could not auto-generate a down migration.
+-- Please write an appropriate down migration for the SQL below:
+-- alter table "public"."workspace_members" add column "created_at" timestamptz
+--  not null default now();
```
Comment (cyborch) [inline hasura/migrations/default/1682141433503_alter_table_public_workspace_members_add_column_created_at/down.sql:4]: Add `alter table "public"."workspace_members" drop column "created_at"` here?

File: hasura/migrations/default/1682141443863_alter_table_public_workspace_members_add_column_updated_at/down.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+--   _new."updated_at" = NOW();
+--   RETURN _new;
+-- END;
+-- $$ LANGUAGE plpgsql;
+-- CREATE TRIGGER "set_public_workspace_members_updated_at"
+-- BEFORE UPDATE ON "public"."workspace_members"
+-- FOR EACH ROW
+-- EXECUTE PROCEDURE "public"."set_current_timestamp_updated_at"();
+-- COMMENT ON TRIGGER "set_public_workspace_members_updated_at" ON "public"."workspace_members"
+-- IS 'trigger to set value of column "updated_at" to current timestamp on row update';
```
Comment (cyborch) [inline hasura/migrations/default/1682141443863_alter_table_public_workspace_members_add_column_updated_at/down.sql:21]: Add
```
DROP FUNCTION "public"."set_current_timestamp_updated_at"();
DROP TRIGGER "set_public_workspace_members_updated_at";
```

File: hasura/migrations/default/1682141457260_alter_table_public_workspace_members_add_column_deleted_at/up.sql [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  _new."deleted_at" = NOW();
+  RETURN _new;
+END;
+$$ LANGUAGE plpgsql;
+CREATE TRIGGER "set_public_workspace_members_deleted_at"
+BEFORE UPDATE ON "public"."workspace_members"
+FOR EACH ROW
+EXECUTE PROCEDURE "public"."set_current_timestamp_deleted_at"();
+COMMENT ON TRIGGER "set_public_workspace_members_deleted_at" ON "public"."workspace_members"
+IS 'trigger to set value of column "deleted_at" to current timestamp on row update';
```
Comment (cyborch) [inline hasura/migrations/default/1682141457260_alter_table_public_workspace_members_add_column_deleted_at/up.sql:19]: Add
```
DROP TRIGGER "set_public_workspace_members_deleted_at";
DROP FUNCTION "public"."set_current_timestamp_deleted_at"();
```

File: hasura/migrations/default/1682337427850_alter_table_public_workspaces_add_column_domain_name/down.sql [hunk 4]
```diff
@@ -0,0 +1,4 @@
+-- Could not auto-generate a down migration.
+-- Please write an appropriate down migration for the SQL below:
+-- alter table "public"."workspaces" add column "domain_name" text
+--  null unique;
```
Comment (cyborch) [inline hasura/migrations/default/1682337427850_alter_table_public_workspaces_add_column_domain_name/down.sql:4]: Add
```
alter table "public"."workspaces" drop column "domain_name";
```

**Review by cyborch — APPROVED** [review]
Looks good to me :)

Consider adding `down` migrations.

**Pull request merged**